python/tank/platform/application.py

```diff
@@ -207,6 +209,17 @@ def log_exception(self, msg):
     ##########################################################################################
     # internal API
 
+    def event(self, event):
```

Comment:
hmm, i realise that we are not following the "protected" paradigm in our code at all. 

Current methods:

```
init_app
change_context
pre_context_change
post_context_change
init_app
post_engine_init
destroy_app
```

All of those methods should never really be called any external code, they should always be called by the system. 

I guess with that in mind, maybe events could be called something like

```
event_engine(event) # evt is always sgtk.platform.events.EngineEvent
event_file_open(event) # evt is always sgtk.platform.events.FileOpenEvent
```

What do you think?